pkg/patch/patch.go

```diff
@@ -50,6 +54,65 @@ var (
 	bkNewClient = buildkit.NewClient
 )
 
+type archDigest struct {
+	tag    string
+	digest string
+	plat   types.PatchPlatform
+}
+
+// archTag returns "patched-arm64" or "patched-arm-v7" etc.
+func archTag(base, arch, variant string) string {
+	if variant != "" {
+		return fmt.Sprintf("%s-%s-%s", base, arch, variant)
+	}
+	return fmt.Sprintf("%s-%s", base, arch)
+}
+
+// dockerImageDigest returns the image digest for a given image name.
+// It uses the "docker image inspect" command to get the digest.
+func dockerImageDigest(img string) (string, error) {
+	out, err := exec.Command("docker", "image", "inspect", "--format={{index .Id}}", img).Output()
```

Comment:
we might have already discussed this, but was there a library we could use for this?

Reply:
I think we did something similar for local media type in the oci mediatype PR, I've changed it to use that library. Happy to use another library if there is one that can do this better!